ugvc/mrd/featuremap_consensus_utils.py

```diff
@@ -0,0 +1,111 @@
+from __future__ import annotations
+
+from collections import defaultdict
+from os.path import basename
+from os.path import join as pjoin
+
+import numpy as np
+import pysam
+
+
+def write_a_pileup_record(
+    record_dict: dict, rec_id: str, out_fh: pysam.VariantFile, header: pysam.VariantHeader, min_qual: int
+):
+    info_fields = [
+        "X_QUAL",
+        "X_RN",
+    ]
+
+    format_fields = ["GT", "AD", "DP", "VAF"]
+
+    rec = header.new_record()
+    rec.chrom = rec_id.split("-")[0]
+    rec.pos = int(rec_id.split("-")[1])
+    rec.ref = rec_id.split("-")[2]
+    rec.alts = [rec_id.split("-")[3]]
+    rec.id = "."
+    rec.qual = np.mean(record_dict["X_QUAL"])
+    record_dict["X_RN"] = "|".join(record_dict["X_RN"])
+    curr_filter = "PASS" if rec.qual >= min_qual else "FAIL"
+    record_dict["VAF"] = record_dict["FILTERED_COUNT"] / record_dict["DP"]
+    rec.filter.add(curr_filter)
+
+    for info_field in info_fields:
+        rec.info[info_field] = record_dict[info_field]
+
+    record_dict["GT"] = (0, 1)
+    record_dict["AD"] = [record_dict["FILTERED_COUNT"], record_dict["DP"] - record_dict["FILTERED_COUNT"]]
+    for format_field in format_fields:
+        rec.samples["SAMPLE"][format_field] = record_dict[format_field]
+
+    out_fh.write(rec)
+    return rec
+
+
+def pileup_featuremap(featuremap: str, output_dir: str, genomic_interval: str, min_qual: int):
+    """
+    Pileup featuremap vcf to a reagular vcf, save the mean quality and read count
```

Comment:
1. Please complete the docstring, including inputs. 
2. Why is genomic_intervals a required input? According to what it says below it should be a default of "", though None is better
3. I suggest a default of min_qual=0